### ISSUE TITLE
Raw parameters in the footer

### ISSUE THREAD

**Issue by mathieuroquefort** [sympa-community/sympa#501]
Hello,

I got complaints from our users because the links in the footer didn't work.
And it appears that the parameters are not replaced by their values.

Here is the footer in the messages :
```
Si vous désirez vous désabonner, envoyez un message en cliquant sur ce lien : mailto:sympa@domain.tld?subject=uns%20[% list.name %]
. 
Si vous désirez modifier votre abonnement, rendez-vous à cette adresse : 
[% conf.wwsympa_url %]/sympa/suboptions/[% list.name %] 
```

It did work previously. I don't understand what happened...

Version
-------
Sympa 6.1.17 / CentOS 6.10

Installation method
-------------------
?

Expected behavior
-----------------
[% list.name %] should be replace by the list name, and [% conf.wwsympa_url %] by the URL of the web interface.

Actual behavior
---------------
Parameters are not replaced, and displayed raw.

Additional information
----------------------

**Comment by ikedas** [comment]
Hi @mathieuroquefort,

From list info page, please go to [Edit list config] - [Sending/receiving setup], and check if [Attachment type] is "append to message body".  If "add a new MIME part" is set, footer will be appended as a separate part and won't be replaced.

**Comment by mathieuroquefort** [comment]
I've just tried it. Now, on the list parameters, footer_type = append

But then :
- If I send an html message, no footer is added
- If I send a plain text message, the footer is appended but the parameters are not replaced

I had already tested different parameters here before.
The other problem is, that almost nobody sends plain text message now. I don't know how to append footer on html messages, and that is a real problem.

**Comment by ikedas** [comment]
I realized that you use a bit earlier version of Sympa.  Please upgrade to recent version --- if possible, 6.2.x, or at least 6.1.25 --- and check if problem will be solved.

**Comment by mathieuroquefort** [comment]
We can't do that easily : we use a managed server, running CentOS. And CentOS hasn't been tested with sympa 6.2.
We use, I think, the latest validated version with CentOS6.10. Our manager doesn't like using non-validated softwares.
Therefore, I'm looking for a more short-term solution. For now on, I have replaced the parameters with their values on each list. With 300 lists that's quite a job, but that was the only solution.

Does the 6.2 version of sympa improve the footer appending in html messages ? I can't find this information in the documentation.

**Comment by ikedas** [comment]
I don't know what is "validation with ContOS" (Sympa 6.1 also have not been validated. Besides).

I think, as you say, only short-term solution without upgrading Sympa is preparing footers by each list.

Feature of personalization (merge) and footer were improved on recent 6.1.x (not only 6.2).  AFAIK appending footer to HTML message is more or less possible.

**Comment by racke** [comment]
The footer type _append_  just adds the file contents and never evaluates it, if I read the source code correctly.
This seems a unnecessary restriction to me.

**Comment by ikedas** [comment]
@racke, personalization (evaluation of template syntax) is done by `bulk.pl`.  As you wrote, `sympa_msg.pl` (`sympa.pl` with 6.1) just adds the files.

**Comment by mathieuroquefort** [comment]
Hello,
I'm coming back here, sorry for the long delay.

The problem is always there :
- If I set footer_type to mime, the footer message is appended but not interpreted.
- If I set footer_type to append, the footer message is not appended. When I open the raw message code, the text is simply absent.

So it looks like the only solution is to set all the footer_types to mime, and to personnalize all the footer messages one by one. We have more than 300 lists, I'll have much fun :(

Does anyone have another solution ?
